### ISSUE TITLE
No response sent if unbind has error

### ISSUE THREAD

**Issue by makandre** [eclipse/codewind#1054]
**Codewind version:** 0.6.0
**OS:**

**Che version:**
**IDE extension version:**
**IDE version:**
**Kubernetes cluster:**

**Description:**

If `unbind` throws an error on this line:

https://github.com/eclipse/codewind/blob/master/src/pfe/portal/routes/projects/remoteBind.route.js#L399

It would go to the `catch` and no response is send back on the `res` object.

**Steps to reproduce:**

**Workaround:**

**Comment by malincoln** [comment]
@makandre pls verify and close. Thanks

**Comment by makandre** [comment]
I have a question about the fix:  

If an error is thrown at `await user.unbindProject(project);` (this happens after a `202 request accepted` is already sent), in that case I think sending a `500` is going to fail

@markcor11 ?

**Comment by markcor11** [comment]
@makandre good catch,  I think setting the status codes or response mid way through a function is generally bad practice for this exact reason. I can refactor that entire route and move to a single response set right at end.

**Comment by markcor11** [comment]
@makandre the changes in 1630 have now been reverted,  looking at the code it does appear that if `await user.unbindProject(project);` throws an exception,  the response back to the user will be via a socket IO message rather than via the REST request. I've discussed this within the squad and that is the intended behaviour which happens in two phases. 

1. Accept or reject the request to unbind a project 
2. Unbind the project

Part one is a rest response and is immediate 
Part two is via a SocketIO emit event when the operation either completes or errors

To see the full status of the unbind operation, a combination of both the REST response and SocketIO message should be monitored. 

https://eclipse.github.io/codewind/#/paths/~1api~1v1~1projects~1{id}~1unbind/post
